examples/complex-predicates/Main.hs

```diff
@@ -37,7 +40,7 @@ catComment :: Scraper String String
 catComment =
     -- 1. First narrow the current context to the div containing the comment's
     --    textual content.
-    chroot ("div" @: [hasClass "comment", hasClass "text"]) $ do
+    chroot (("div" :: TagName String) @: [hasClass "comment", hasClass "text"]) $ do
```

Comment:
I think this is why I chose to mix String and StringLike. It seems a lot less convenient to have to explicitly type each tag name as a `TagName String`.

Is there a specific benefit that using StringLike enables?

Reply:
What if we have a String or Text specific file that just declares the type

Reply:
Hrmmm, there's already Text.HTML.Scalpel and Text.HTML.Scalpel.Core. I'd rather not create more re-exports of the API if its not necessary.

Other than API consistency is there a benefit to having StringLike, Text, and String versions of the API? I'm not super familiar with the implementation of OverloadedStrings, but I think the conversion happens at run time. If that's the case I don't think there would be a performance difference since internally, the library is already converting to `Text` for comparing nodes.

Reply:
I am using libs that use Text and it's inconvenient to convert to strings and from strings.

What if we have a classed based file Text.HTML.Scalpel.StringLike and then just apply String to all the functions in Text.HTML.Scalpel?

Reply:
Gotcha, I was not really expecting people to be passing dynamic values as attributes. I think having a single `Text.HTML.Scalpel.StringLike` is more palatable, let's go with that.